backends/tfhe-cuda-backend/cuda/src/integer/bitwise_ops.cuh

```diff
@@ -14,15 +14,53 @@
 template <typename Torus>
 __host__ void host_integer_radix_bitop_kb(
     cudaStream_t const *streams, uint32_t const *gpu_indexes,
-    uint32_t gpu_count, Torus *lwe_array_out, Torus const *lwe_array_1,
-    Torus const *lwe_array_2, int_bitop_buffer<Torus> *mem_ptr,
-    void *const *bsks, Torus *const *ksks, uint32_t num_radix_blocks) {
+    uint32_t gpu_count, CudaRadixCiphertextFFI *lwe_array_out,
+    CudaRadixCiphertextFFI const *lwe_array_1,
+    CudaRadixCiphertextFFI const *lwe_array_2, int_bitop_buffer<Torus> *mem_ptr,
+    void *const *bsks, Torus *const *ksks) {
 
   auto lut = mem_ptr->lut;
+  uint64_t degrees[lwe_array_1->num_radix_blocks];
+  if (mem_ptr->op == BITOP_TYPE::BITAND) {
```

Comment:
This make reading harder. What if you encapsulate this conditional?

Reply:
What do you mean by encapsulate? Create a function to update noise/degree for bitops/scalar bitops you mean?

Reply:
Yes.
I would keep the conditional inside `host_integer_radix_bitop_kb` but for each case I would call a function like the `after_X()` methods we have at the rust side.